modules/apps/transfer/internal/convert/convert.go

```diff
@@ -0,0 +1,57 @@
+package convert
+
+import (
+	"strings"
+
+	"github.com/cosmos/ibc-go/v8/modules/apps/transfer/internal/denom"
+	v1types "github.com/cosmos/ibc-go/v8/modules/apps/transfer/types"
+	v3types "github.com/cosmos/ibc-go/v8/modules/apps/transfer/types/v3"
+)
+
+// PacketDataV1ToV3 converts a v1 (ICS20-V1) packet data to a v3 (ICS20-V2) packet data.
+func PacketDataV1ToV3(packetData v1types.FungibleTokenPacketData) v3types.FungibleTokenPacketData {
```

Comment:
I don't really like how we're  going from V1 -> V3 in terms of proto, but going from ICS20-V1 to V2, open to suggestions here.

Reply:
i'm ok with the `ICS20V1ToV2` framing as opposed to V1 -> V3 for PacketData!

Reply:
we can do this in a followup pr

Reply:
Here are some potential ways to improve naming:
- Move v2 to v1 and v3 to v2.
- The name of the struct doesn't really mean much. So we could Name it `FungibleTokenPacketDataV2` and put it in v2.
- Move directly to `ics20-3`, SDK `v0.52` style.
- Keep as is

Reply:
I suggested moving the proto definition to proto pkg v3 to see what it would look like seeing as it kind of makes sense to follow some proto rules. But maybe adding `FunginbleTokenPacketDataV2` or some other option is less noisy given that we are already on proto pkg v2.

When proto pkg v2 was added for packet data with amount as string we removed the v1 proto definition ([ref](https://github.com/cosmos/ibc-go/pull/350/files#diff-b92045f77047ee77e807794e59ed0ee6486f96087f31990086202a053df40c9e)), but this scenario is different given we want to maintain the existing packet data in parallel. 

There is a few different options here as @srdtrk has pointed out. I think we should avoid `ics20-3` at all costs - this would add too much confusion 😅 
I think another option is probably making the new packet data the proto pkg v2 `FungibleTokenPacketData` and renaming the existing one to have the V1 suffix or "Legacy" prefix or something. That might be a cleaner option completely, rather than introducing proto pkg v3 - I think this should be possible but will need to look into it a bit more, any thoughts on this?

Reply:
```I think another option is probably making the new packet data the proto pkg v2 FungibleTokenPacketData and renaming the existing one to have the V1 suffix or "Legacy" prefix or something. ```

this makes the most sense to me, as I think having the proto pkg move from v1 -> v2 in parallel with ics20 moving from v1 -> v2 seems to make the most sense. If there is general agreement I can make this update in this  pr!

Reply:
https://github.com/cosmos/ibc-go/issues/6170 added this issue to track this topic so we can continue on in parallel!